docker/cli/Dockerfile

```diff
@@ -0,0 +1,16 @@
+ARG BASE_ZKVM_IMAGE_TAG=ere-base-zkvm:latest
+
+FROM ${BASE_ZKVM_IMAGE_TAG}
+
+COPY . /ere
+
+WORKDIR /ere
+
+ARG ZKVM
+
+RUN cargo build --release --package ere-cli --bin ere-cli --features ${ZKVM} && \
+    cp /ere/target/release/ere-cli /ere/ere-cli && \
+    cargo clean && \
+    rm -rf $CARGO_HOME/registry/src $CARGO_HOME/registry/cache
+
+ENTRYPOINT ["/ere/ere-cli"]
```

Comment:
Re https://github.com/eth-act/ere/pull/66#discussion_r2243022237 and https://github.com/eth-act/ere/pull/66#discussion_r2242788242 now it's a `Dockerfile` and looks much cleaner, previously there are more args so I decided to have a builder, but in the end it seems we only need to select `ZKVM`